include/sempr/entity/spatial/Geometry.hpp

```diff
@@ -49,36 +60,46 @@ class Geometry : public Entity {
     */
     void transformToCS(SpatialReference::Ptr cs);
 
-    /** Exception that may be thrown during transformToCS. */
-    class TransformException : public std::exception {
-    private:
-        std::string message_;
-    public:
-        explicit TransformException(const std::string& m) : message_(m) {}
-        virtual const char* what() const throw() {
-            return message_.c_str();
-        }
-    };
+    // apply a read only filter to find the min and max values of the geometry in his local coordinate system in 3d.
+    void findEnvelope(geom::Coordinate& min, geom::Coordinate& max);
 
     /**
         Get a new entity with the same geometry (copy) referring to the same instance of
         SpatialReference. (implemented by raw_clone)
     */
     Geometry::Ptr clone() const;
 
-private:
+    static geom::Geometry* importFromWKB(const std::basic_string<char> &buffer);
+    static std::basic_string<char> exportToWKB(const geom::Geometry *geom);
+
+    static geom::Geometry* importFromWKT(const std::string &text);
+    static std::string exportToWKT(const geom::Geometry *geom);
+
+  protected:
+    static const geom::GeometryFactory* factory_;
+
+    /**
+     * @brief Get the GEOS GEOM Geometry object as mutable (non-const) pointer.
+     */
+    virtual geom::Geometry* getGeometryMut() const;
```

Comment:
Isn't it self-contradictory to mark a method that returns a pointer to non-const and is explicitly called `getGeometryMut[able]` as `const`?! ^^

Reply:
This is more a code style question :-)
I have mark this method as const because it is a Getter and the method itself will not change anything of the current object. ( and I always like to have a Getter const)
But the object can (not have to) be changed by the returning pointer. So I see the self-confradictory.

On the other side you will not call getGeometry**Mut** if you didn't want to change something. Than you also can call getGeometry even if you are in a const method.

Reply:
Its clean in b592d1b8d82f928243794dfc865afae4017728ab

Reply:
Yeah, maybe that's just not totally straight-forward... IMHO if I'm not allowed to modify a `entity::Geometry` (i.e., I've only got a `const entity::Geometry` to work with), I should not be able to get access to the non-const `geos::geom::Geometry`, as changing that would in fact modify `entity::Geometry` as it only acts as a wrapper.

But since you already changed that I'm fine. :)